server/src/main/scala/com/cloudera/livy/LivyConf.scala

```diff
@@ -245,15 +245,25 @@ class LivyConf(loadDefaults: Boolean) extends ClientConf[LivyConf](null) {
   /** Return the spark deploy mode Livy sessions should use. */
   def sparkDeployMode(): Option[String] = Option(get(LIVY_SPARK_DEPLOY_MODE)).filterNot(_.isEmpty)
 
+  /** Return the  spark home version */
+  def SPARK_HOME_VER(version: String): Option[String] = {
+    Option(get(s"livy.server.spark-home-$version"))
+      .orElse(throw new IllegalArgumentException(
+        s"Spark version: $version is not supported"))
+  }
   /** Return the location of the spark home directory */
-  def sparkHome(): Option[String] = Option(get(SPARK_HOME)).orElse(sys.env.get("SPARK_HOME"))
+  def sparkHome(version: Option[String] = None): Option[String] = {
+    version.map {version => SPARK_HOME_VER(version)
```

Comment:
On a second look it may be better to use a `case` here, assuming the changes I recommend above something like:
```
case Some(version) => Option(get(SPARK_HOME_VER(version))).orElse(throw ...)
case None => Option(get(SPARK_HOME)).orElse(sys.env.get("SPARK_HOME"))
```
where the None case is just the old code. This would be easier to read overall.